compensated-ruby/lib/compensated/gumroad/event_parser.rb

```diff
@@ -9,27 +9,44 @@ def parses?(request)
         keys.include?(:seller_id) && keys.include?(:product_id) && keys.include?(:product_permalink) && request.data["product_permalink"].include?("gum.co")
       end
 
-      def normalize(data_or_body)
-        data = data_or_body.respond_to?(:key) ? data_or_body : data_from_string(data_or_body)
+      def transform(data_or_body)
+        data = extract(data_or_body)
         body = data_or_body.respond_to?(:key) ? nil : read_and_rewind(data_or_body)
         {
           raw_body: body,
-          raw_event_type: data["resource_name"].to_sym,
+          raw_event_type: data[:resource_name].to_sym,
           raw_event_id: nil,
           payment_processor: :gumroad,
           amount: {
-            paid: data["price"].to_i,
-            currency: data["currency"].upcase,
+            paid: data[:price].to_i,
+            currency: data[:currency].upcase,
           },
           customer: {
-            id: data["purchaser_id"].to_s,
-            email: data["email"],
-            name: data["full_name"],
+            id: data[:purchaser_id].to_s,
+            email: data[:email],
+            name: data[:full_name],
           },
-          timestamp: DateTime.parse(data["sale_timestamp"]),
+          timestamp: DateTime.parse(data[:sale_timestamp]),
         }.compact
       end
 
+      # <b>DEPRECATED:</b> Please use <tt>transform</tt> instead.
+      def normalize(data_or_body)
+        warn '[DEPRECATION] `normalize` is deprecated.  Please use `transform` instead.'
+        transform(data_or_body)
+      end
+
+      def extract(data_or_body)
+        if data_or_body.respond_to?(:key)
+          data_or_body
+        else
+          data_from_string(data_or_body)
+        end.inject({}) do |hsh,(key,value)|
+          hsh[key.to_sym] = value
+          hsh
+        end
+      end
```

Comment:
I was trying to use `Hash#transform_keys` but then Ruby 2.4.x doesn't support it.
Is there a way to check against ruby version so I can do the following?
```ruby
if ruby version >= 2.5
  data.transform_keys(&:to_sym)
else
  data.inject({}) do |hsh,(key,value)|
    hsh[key.to_sym] = value
    hsh
  end
end
```

Reply:
You could use Duck Typing with`respond_to?`:  `if data.respond_to?(:transform_keys)`